src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -229,6 +293,26 @@ private String getInsertSql() {
     }
   }
 
+  private String getDeleteSql() {
+    String sql = null;
+    if (config.deleteEnabled) {
+      switch (config.pkMode) {
+        case RECORD_KEY:
+          if (fieldsMetadata.keyFieldNames.isEmpty()) {
+            throw new ConnectException("Require primary keys to support delete");
+          }
+          sql = dbDialect.buildDeleteStatement(
+              tableId,
+              asColumns(fieldsMetadata.keyFieldNames)
+          );
+          break;
+        default:
+          throw new ConnectException("Deletes are only supported for pk.mode record_key");
```

Comment:
Strictly speaking, this case wouldn't be necessary if Config validation ensures that `config.pkMode == RecordKey` when `config.deleteEnabled`. However, I still think it's good to have, just in case.